### ISSUE TITLE
Implement Non-interactive defaults

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-core#455]
## Summary

Use the non-interactive defaults described in the [spec](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#non-interactive-default-rules)

## Details

The spec describes the non-interactive default rules, which allow for commitments to a message to be generated without external info. Please see [spec](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#non-interactive-default-rules) for more detail. 

The app will also need to import this functionality, so that it can properly arrange shares while staying within the max square size.

This issue should only be handled after determining square size #454 has been implemented.

### Action Items
- [ ] Add non-default functionality in celestia-core's (probably around [message splitting](https://github.com/lazyledger/lazyledger-core/blob/b8e72df9d005e5f19b6104adf7bc3c62e616eb6d/types/share_splitting.go#L28-L52))
- [ ] fix incorrectly generated commitments in the app [#53](https://github.com/celestiaorg/lazyledger-app/issues/53)
- [ ] import functionality to celestia-app
- [ ] the app needs to add the starting location of each message as meta data to `SignedTransactionDataPayForMessage`s
- [ ] add non-interactive default psuedo-code to the spec ([comment](https://github.com/celestiaorg/lazyledger-core/issues/455#issuecomment-879026544))

blocked by #454

**Comment by liamsi** [comment]
While at it, we should contribute back to the spec to contain these "Non-interactive defaults" as pseudo-code. The current English description is too difficult to follow and predestined for pseudo-code.

**Comment by evan-forbes** [comment]
personal note to myself to not forget that we add the length delimiter to messages and need to account for this

https://github.com/celestiaorg/celestia-core/blob/1609bde2a41604fff1ebcb202e0e9cce1048fb3b/types/shares.go#L47-L54)

**Comment by evan-forbes** [comment]
closing in favor of https://github.com/celestiaorg/celestia-app/issues/382
